### PR TITLE
 OCPBUGS-13621: Fix singular Ingress and API cluster VIPs removal

### PR DESCRIPTION
**Pull request by nmagnezi**
This change fixes a case where the user attempts to remove Ingress and API VIPs. Cleanup used to fail due to a missing deletion as a part of the deletion db transaction.

Added unit tests to verify all deletion combination works for singular, plural, and a combination of VIP params.

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [x] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [x] Automation (CI, tools, etc)
- [x] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by nmagnezi** [comment]
/hold 
still WIP, but need to see CI in action

File: internal/network/utils.go [hunk 1]
```diff
@@ -374,6 +380,12 @@ func UpdateVipsTables(db *gorm.DB, cluster *common.Cluster, apiVipUpdated bool,
 
 	if apiVipUpdated {
 		if err := db.Transaction(func(tx *gorm.DB) error {
+			if err := db.Model(&models.Cluster{}).Where("id = ?", *cluster.ID).
```
Comment (filanov) [inline internal/network/utils.go:383]: if you are using update do you still need the delete below? 
same comment for the `ingress_vip`

Comment (filanov) [reply internal/network/utils.go:383]: actually the delete is for the table, ignore

**Review by filanov — APPROVED** [review]

**Pull request merged**